ocs_ci/ocs/couchbase.py

```diff
@@ -165,42 +139,89 @@ def test_couchbase_workload_simple(self, pillowfight):
+        to bringing up the operator.  Secrets, rolebindings and serviceaccounts
+        need to also be generated.
+
+        Once the couchbase operator is running, we need to wait for the three
+        worker pods to also be up.  Then a pillowfight task is started.
+
+        After the pillowfight task has finished, the log is collected and
+        analyzed.
+
+        Raises:
+            Exception: If pillowfight results indicate that a minimum performance
+                level is not reached (1 second response time, less than 1000 ops
+                per second)
+
+        """
+        logging.info('Running pre reqs')
+        self._prereq()
+        logging.info('Creating pods..')
         cb_example = templating.load_yaml(constants.COUCHBASE_WORKER_EXAMPLE)
+        cb_example['spec']['servers'][0]['size'] = replicas
         self.cb_examples = OCS(**cb_example)
         self.cb_examples.create()
+
         # Wait for last of three workers to be running.
-        for cb_wrk_pod in TimeoutSampler(
+
+        logging.info('Waiting for the pods to Running')
+        for cb_wrk_pods in TimeoutSampler(
             self.WAIT_FOR_TIME,
             3,
             get_pod_name_by_pattern,
-            'cb-example-0002',
-            self.COUCHBASE_OPERATOR
+            'cb-example',
+            constants.COUCHBASE_OPERATOR
         ):
             try:
-                if self.is_up_and_running(cb_wrk_pod[0], self.up_check):
-                    # once last pod is up, make sure all are ready
+                if len(cb_wrk_pods) == replicas:
                     counter = 0
-                    for wpodn in range(0, 3):
-                        cbw_pod = f"cb-example-{wpodn:04}"
-                        if self.is_up_and_running(cbw_pod, self.up_check):
+                    for cb_pod in cb_wrk_pods:
+                        if self.is_up_and_running(cb_pod, self.up_check):
                             counter += 1
-                    if counter == 3:
+                            logging.info(f'Couchbase worker {cb_pod} is up')
+                    if counter == replicas:
                         break
             except IndexError:
-                log.info("Couchbase workers are not up")
+                logging.info(
+                    f'Expected number of couchbase pods are {replicas} '
+                    f'but only found {len(cb_wrk_pods)}'
+                )
 
-        pillowfight.run_pillowfights()
-        pillowfight.analyze_all()
+    def run_workload(self, replicas):
+        """
+        Running workload with pillow fight operator
+        Args:
+         replicas (int): Number of pods
+
+        """
+        logging.info('Running IOs...')
+        PillowFight.run_pillowfights(self, replicas=replicas)
+
+    def analyze_run(self):
+        """
+        Analyzing the workload run logs
+
+        """
+        logging.info('Analyzing  workload run logs..')
+        PillowFight.analyze_all(self)
```

Comment:
you are writing all the replica's of pillowfight pods to the same file?
If yes, how will you differentiate which pod ouput is what?

Reply:
Yes, the same file only. I'll work on it, to make different log files to analyze with loop